### PR TITLE
Add a way to deep copy an object recursively

### PR DESCRIPTION
**Pull request by ptbhatcoder**
For [Q3](https://github.com/sadanandpai/javascript-code-challenges/blob/main/challenges/objects-challenges.md#q3) in functions challenges

### CODE REVIEW

File: challenges/objects-challenges.md [hunk 1]
```diff
@@ -90,6 +90,25 @@ const x = { ...obj};
...
 ```
 
+```js
+function deepCopy(obj){
+   if(!obj) return obj;
+   const copyObj = {};
+   for(const key in obj){
+      let val;
+      if(obj[key] !== 'object')
+        nVal[key] = obj[key];
```
Comment (sadanandpai) [inline challenges/objects-challenges.md:100]: nVal is not declared

File: challenges/objects-challenges.md [hunk 2]
```diff
@@ -90,6 +90,25 @@ const x = { ...obj};
 const x = JSON.parse(JSON.stringify(obj));
 ```
 
+```js
+function deepCopy(obj){
+   if(!obj) return obj;
+   const copyObj = {};
+   for(const key in obj){
+      let val;
+      if(obj[key] !== 'object')
```
Comment (sadanandpai) [inline challenges/objects-challenges.md:99]: typeof is missing

File: challenges/objects-challenges.md [hunk 3]
```diff
@@ -90,6 +90,25 @@ const x = { ...obj};
...
+        nVal[key] = obj[key];
+      else 
+        copyObj[key] = deepCopy(obj[key]);
+      
+      copyObj[key] = val;
```
Comment (sadanandpai) [inline challenges/objects-challenges.md:104]: is there any use of val

**Review by sadanandpai — CHANGES_REQUESTED** [review]

File: challenges/objects-challenges.md (hunk 1 above)
Comment (ptbhatcoder) [reply challenges/objects-challenges.md]: Fixed. It was `copyObj`

File: challenges/objects-challenges.md (hunk 2 above)
Comment (ptbhatcoder) [reply challenges/objects-challenges.md]: Fixed

**Review by sadanandpai — APPROVED** [review]

**Comment by sadanandpai** [comment]
Solution does not work properly if arrays are present inside

**Pull request merged**